engine/table/src/main/java/io/deephaven/engine/table/impl/sources/regioned/RegionedColumnSourceManager.java

```diff
@@ -812,21 +830,34 @@ public void pushdownFilter(
                             input.subSetByKeyRange(locationStartKey, locationEndKey)) {
                         locationInput.shiftInPlace(-locationStartKey); // Shift to the region's key space
                         locationInput.retain(entry.rowSetAtLastUpdate); // intersect in place with region's row set
-                        try (final PushdownResult result = entry.location.pushdownFilter(columnSourceMap, filter,
-                                locationInput, usePrev, context, costCeiling)) {
-                            // Add the results to the global set.
-                            synchronized (match) {
-                                match.insertWithShift(locationStartKey, result.match());
-                            }
-                            synchronized (maybeMatch) {
-                                maybeMatch.insertWithShift(locationStartKey, result.maybeMatch());
+
+                        final Consumer<PushdownResult> resultConsumer = result -> {
+                            try (final PushdownResult ignored = result) {
+                                // Add the results to the global set.
+                                synchronized (match) {
+                                    match.insertWithShift(locationStartKey, result.match());
+                                }
+                                synchronized (maybeMatch) {
+                                    maybeMatch.insertWithShift(locationStartKey, result.maybeMatch());
+                                }
                             }
-                        } catch (final Exception e) {
-                            // If we have an error, we need to notify the user and stop processing.
-                            onError.accept(e);
-                        }
+                        };
+                        entry.location.pushdownFilter(filter, locationInput, fullSet, usePrev, context, costCeiling,
+                                jobScheduler, resultConsumer, onError);
                     }
                     locationResume.run();
                 }, () -> onComplete.accept(PushdownResult.of(match, maybeMatch)), onError);
     }
+
+    /**
+     * Simple implementation of the abstract {@link BasePushdownFilterContext}. Can be extended to add additional state
+     * if needed.
+     */
+    public static class RegionedColumnSourcePushdownFilterContext extends BasePushdownFilterContext {
```

Comment:
If we don't need why not return the base?

Reply:
only because I made `BasePushdownFilterContext` abstract, will change to a concrete class.